dataprofiler/profilers/profile_builder.py

```diff
@@ -504,7 +511,33 @@ def _update_row_statistics(self, data):
         self.hashed_row_dict = dict.fromkeys(
             pd.util.hash_pandas_object(data, index=False), True
         )
-        self.null_in_row_count = data.isnull().any(axis=1).sum()
+
+        # Calculate Null Column Count
+        null_rows = {}
+        null_in_row_count = {}
+        first_col_flag = True
+        for column in self.profile:
+            if "statistics" in self.profile[column].profile and \
+                    "null_types_index" in self.profile[column]\
+                    .profile["statistics"]:
+
+                # Cycle through each column and get all the null indices 
+                null_type_dict = self.profile[column].profile["statistics"]\
+                                 ["null_types_index"]
+                null_row_indices = {item for sublist in null_type_dict.values()
+                                    for item in sublist}
+
+                # Find the common null indices between the columns
+                if first_col_flag:
+                    null_rows = null_row_indices
+                    null_in_row_count = null_row_indices
+                    first_col_flag = False
+                else:
+                    null_rows = null_rows.intersection(null_row_indices)
+                    null_in_row_count = null_in_row_count.union(null_row_indices)
```

Comment:
technically we have a function which does this called `_combine_unique_sets` which is supposed to be faster I believe.

Reply:
thats for lists, kind of ironic name I think